sdk/ai/azure-ai-projects/azure/ai/projects/models/_patch.py

```diff
@@ -0,0 +1,1372 @@
+    :ivar connection_type: The connection type .
+    :vartype connection_type: ~azure.ai.projects.models._models.ConnectionType
+    :ivar endpoint_url: The endpoint URL associated with this connection
+    :vartype endpoint_url: str
+    :ivar key: The api-key to be used when accessing the connection.
+    :vartype key: str
+    :ivar token_credential: The TokenCredential to be used when accessing the connection.
+    :vartype token_credential: ~azure.core.credentials.TokenCredential
+    """
+
+    def __init__(
+        self,
+        *,
+        connection: GetConnectionResponse,
+        token_credential: Union[TokenCredential, AsyncTokenCredential, None] = None,
+    ) -> None:
+        self.id = connection.id
+        self.name = connection.name
+        self.authentication_type = connection.properties.auth_type
+        self.connection_type = cast(ConnectionType, connection.properties.category)
+        self.endpoint_url = (
+            connection.properties.target[:-1]
+            if connection.properties.target.endswith("/")
+            else connection.properties.target
+        )
+        self.key: Optional[str] = None
+        if hasattr(connection.properties, "credentials"):
+            if hasattr(connection.properties.credentials, "key"):
+                self.key = connection.properties.credentials.key
+        self.token_credential = token_credential
+
+    def to_evaluator_model_config(self, deployment_name, api_version) -> Dict[str, str]:
+        connection_type = self.connection_type.value
+        if self.connection_type.value == ConnectionType.AZURE_OPEN_AI:
+            connection_type = "azure_openai"
+
+        if self.authentication_type == "ApiKey":
+            model_config = {
+                "azure_deployment": deployment_name,
+                "azure_endpoint": self.endpoint_url,
+                "type": connection_type,
+                "api_version": api_version,
+                "api_key": f"{self.id}/credentials/key",
+            }
+        else:
+            model_config = {
+                "azure_deployment": deployment_name,
+                "azure_endpoint": self.endpoint_url,
+                "type": self.connection_type,
+                "api_version": api_version,
+            }
+        return model_config
+
+    def __str__(self):
+        out = "{\n"
+        out += f' "name": "{self.name}",\n'
+        out += f' "id": "{self.id}",\n'
+        out += f' "authentication_type": "{self.authentication_type}",\n'
+        out += f' "connection_type": "{self.connection_type}",\n'
+        out += f' "endpoint_url": "{self.endpoint_url}",\n'
+        if self.key:
+            out += f' "key": "REDACTED"\n'
+        else:
+            out += f' "key": null\n'
+        if self.token_credential:
+            out += f' "token_credential": "REDACTED"\n'
+        else:
+            out += f' "token_credential": null\n'
+        out += "}\n"
+        return out
+
+
+class SASTokenCredential(TokenCredential):
```

Comment:
Should there be an async version of this?

Reply:
@kristapratico yes, likely! Thanks, I'm added a comment in the code, and also created a task to look into this. Since this is not yet supported by the service and not required for Ignite, I will resolve this comment for now.